pkg/api/v1alpha1/cluster_webhook.go

```diff
@@ -134,10 +134,12 @@ func validateImmutableFieldsCluster(new, old *Cluster) field.ErrorList {
 			field.Invalid(field.NewPath("spec", "GitOpsRef"), new.Spec.GitOpsRef, "field is immutable"))
 	}
 
-	if !RefSliceEqual(new.Spec.IdentityProviderRefs, old.Spec.IdentityProviderRefs) {
-		allErrs = append(
-			allErrs,
-			field.Invalid(field.NewPath("spec", "IdentityProviderRefs"), new.Spec.IdentityProviderRefs, "field is immutable"))
+	for _, identityProvider := range new.Spec.IdentityProviderRefs {
+		if identityProvider.Kind == AWSDatacenterKind {
```

Comment:
what is this check doing? why checking if the identity provider kind is `AWSDatacenterConfig`? we dont support aws provider. shouldn't it be checking the awsiamconfig?

Reply:
yeah, good catch, updating!